src/openeo_test_suite/lib/workflows/io.py

```diff
@@ -11,16 +11,21 @@ def load_netcdf_dataarray(
     band_dim_name: str = "bands",
 ) -> xarray.DataArray:
     """Load a data cube from a NetCDF file as a xarray DataArray"""
+
+    # TODO: avoid the try-except, and just work from `open_dataset`?
     try:
-        # TODO: avoid the try-except, and just work from `open_dataset`?
-        data = xarray.open_dataarray(path)
-    except ValueError:
         data = xarray.open_dataset(path, decode_coords="all")
         # VITO/CDSE write a Dataset even if there are no bands, with a default name 'var'
         if len(data.data_vars) == 1 and [v for v in data.data_vars] == ["var"]:
             data = data["var"]
+        # EODC writes a Dataset even if there are no bands, with a default name 'name'
+        elif len(data.data_vars) == 1 and [v for v in data.data_vars] == ["name"]:
+            data = data["name"]
         else:
-            data = data.to_dataarray(dim=band_dim_name)
+            data = data.to_array(dim=band_dim_name)
+    except ValueError:
+        data = xarray.open_dataarray(path)
```

Comment:
If you switch to trying `open_dataset` first, is it actually still necessary to do the try-except? As far as I know, `open_dataset` will also work on data arrays, so doing `open_dataarray` on `except`  might be useless

Reply:
I did not implement the initial `try-except` code, so I am not sure to be honest. For eodc's results it should not be required.

Reply:
these try-except constructs were originally introduced by @clausmichele  I think, e.g. like in c8b6f519849b51606313b0085892551dfa44c87c  so maybe he can comment if the try-except can be dropped if `open_dataset` will be used from first time

Reply:
Yeah, on our side we still use DataArray most of the times, that's why I was checking that.